src/confluent_kafka/src/Admin.c

```diff
@@ -454,19 +455,19 @@ static PyObject *Admin_create_topics (Handle *self, PyObject *args,
                 }
 
                 if (newt->replica_assignment) {
-                        if (newt->replication_factor != -1) {
+                        if (newt->replication_factor != -1 || newt->num_partitions !=-1) {
                                 PyErr_SetString(PyExc_ValueError,
-                                                "replication_factor and "
+                                                "replication_factor/num_partitions and "
                                                 "replica_assignment are "
                                                 "mutually exclusive");
                                 i++;
                                 goto err;
                         }
-
+                        int partitions = sizeof(newt->replica_assignment)/sizeof(newt->replica_assignment[0]);
```

Comment:
unsure what you are trying to do here, i don't think you need to do anything.

Reply:
This only works for statically sized arrays, e.g., `int foo[4];`, in fact NewTopic->replica_assignment isn't an array at all but a single PyObject that represents a Python list.

Also; variable declarations are only allowed at the beginning of scopes and blocks, to support older MSVC compilers.

Reply:
Noted , used Python length function in C API for that!